[EDU-6046] Create Azion Lib docs in English (#1479)
aziontech/docs: src/content/docs/en/pages/devtools/azion-lib/usage/purge.mdx

---
title: Azion `Purge` library
description: >-
  Learn how to use the `Purge` library to manage cache purging in your web applications.
permalink: /documentation/products/azion-lib/purge/
meta_tags: >-

>   Azion Lib, edge computing, edge applications, Azion products

Reviewer: ```suggestion
  Azion Lib, edge computing, edge applications, Azion products, edge cache purge
```

namespace: documentation_azion_lib_purge
---

The **Purge Library** provides methods to interact with the **Edge Cache Purge** API, allowing you to purge URLs, Cache Keys, and Wildcard expressions from the cache. This client is configurable and supports both debug mode and environment variable-based configuration.

You can interact with the API using a `client` or calling the methods directly from the library. When making direct calls, you can use the environment variables to configure the client without passing the token and debug parameters directly.

This is an example of how a `.env` file with your environment variables may look like:

```sh
AZION_TOKEN=<your-api-token>
AZION_DEBUG=true
```

| Variable | Description |
|----------|-------------|
| `AZION_TOKEN` | Your Azion API token. |
| `AZION_DEBUG` | Enable debug mode (true/false). |

:::tip
Setting `AZION_DEBUG` to `true` enables **Debug mode**. This mode provides detailed logging of the API requests and responses.
:::

> If you want to create a specific `client` for interacting with **Purge**, create one by calling the `createClient` method from the library:

Reviewer: ```suggestion
If you want to create a specific `client` for interacting with **Purge**, do it by calling the `createClient` method from the library:
```

const client: AzionPurgeClient = createClient({ token: 'your-api-token', options: { debug: true } });

const { data: purgeURLResponse, error }: AzionPurgeResponse<AzionPurge> = await client.purgeURL([
  'http://www.domain.com/path/image.jpg',
]);
if (purgeURLResponse) {
  console.log('Purge successful:', purgeURLResponse);
} else {
  console.error('Purge failed', error);
}
```

The `createClient` method has the following **parameters** and **return value**:

**Parameters:**

| Parameter | Type | Description |
|-----------|------|-------------|
| `config` | `Partial<{ token: string; options?: OptionsParams }>` | Configuration options for the Domain client. |

**Returns:**

| Return Type | Description |
|-------------|-------------|
| [`AzionDomainsClient`](#aziondomainsclient) | An object with methods to interact with Domains. |

:::note
In the following examples, the methods are called directly without the creation of a `client`. For more information on how to interact with services and products using a `client`, check the [Azion Lib Client documentation](/en/documentation/products/azion-lib/client/).
:::

> 
> ## Usage

Reviewer: ```suggestion

---

## Usage
```

const url: string[] = ['http://www.domain.com/path/image.jpg'];
const { data: response, error }: AzionPurgeResponse<AzionPurge> = await purgeURL(url, { debug: true });
if (response) {
  console.log('Purge successful:', response);
} else {
  console.error('Purge failed', error);
}
```

**Parameters:**

| Parameter | Type | Description |
|-----------|------|-------------|
| `url` | `string[]` | URL(s) to purge. |
| `options` | `AzionClientOptions` (optional) | Client options including debug mode. |

**Returns:**

| Return Type | Description |
|-------------|-------------|
| `Promise<AzionPurgeResponse<AzionPurge>>` | The purge response or error if the purge failed. |

### `purgeCacheKey`

Purges a Cache Key from the Edge Cache.

Example:

```typescript
import { purgeCacheKey } from 'azion/purge';
import type { AzionPurgeResponse, AzionPurge } from 'azion/purge';

const cacheKey: string[] = ['http://www.domain.com/path/image.jpg'];
const { data: response, error }: AzionPurgeResponse<AzionPurge> = await purgeCacheKey(cacheKey, { debug: true });
if (response) {
  console.log('Purge successful:', response);
} else {
  console.error('Purge failed', error);
}
```

> **Parameters:**
> 
> | Parameter | Type | Description |
> |-----------|------|-------------|
> | `cacheKey` | `string[]` | Cache Key(s) to purge. |
> | `options` | `AzionClientOptions` (optional) | Client options including debug mode. |
> 
> **Returns:**

Reviewer: ```suggestion
**Parameters**:

| Parameter | Type | Description |
|-----------|------|-------------|
| `cacheKey` | `string[]` | Cache Key(s) to purge. |
| `options` | `AzionClientOptions` (optional) | Client options including debug mode. |

**Returns**:
```

| Return Type | Description |
|-------------|-------------|
| `Promise<AzionPurgeResponse<AzionPurge>>` | The purge response or error if the purge failed. |

### `purgeWildCard`

Purges using a wildcard expression from the Edge Cache.

Example:

```typescript
import { purgeWildCard } from 'azion/purge';
import type { AzionPurgeResponse, AzionPurge } from 'azion/purge';

const wildcard: string[] = ['http://www.domain.com/path/image.jpg*'];
const { data: response, error }: AzionPurgeResponse<AzionPurge> = await purgeWildCard(wildcard, { debug: true });
if (response) {
  console.log('Purge successful:', response);
} else {
  console.error('Purge failed', error);
}
```

> **Parameters:**
> 
> | Parameter | Type | Description |
> |-----------|------|-------------|
> | `wildcard` | `string[]` | Wildcard expression(s) to purge. |
> | `options` | `AzionClientOptions` (optional) | Client options including debug mode. |
> 
> **Returns:**

Reviewer: ```suggestion
**Parameters**:

| Parameter | Type | Description |
|-----------|------|-------------|
| `wildcard` | `string[]` | Wildcard expression(s) to purge. |
| `options` | `AzionClientOptions` (optional) | Client options including debug mode. |

**Returns**:
```

| Return Type | Description |
|-------------|-------------|
| `Promise<AzionPurgeResponse<AzionPurge>>` | The purge response or error if the purge failed. |

> 
> ## Types

Reviewer: ```suggestion

---

## Types
```
